cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -440,6 +433,41 @@ fn check_internet_connection() -> bool {
     false
 }
 
+// Appends the contents of a file to another file, separated by a delimiter
+fn append_contents(from: &Path, to: &Path) -> Result<(), Error> {
+    let mut from_file = File::open(from)?;
+    let mut from_content = String::new();
+    from_file.read_to_string(&mut from_content)?;
+
+    let mut to_file = OpenOptions::new().read(true).append(true).open(to)?;
+    let mut to_content = String::new();
+    to_file.read_to_string(&mut to_content)?;
+
+    let delimiter = get_merged_file_delimiter(to)?;
+    // if the to file already contains the delimiter, we don't need to append the contents again
+    if to_content.contains(&delimiter) {
+        return Ok(());
+    }
+
```

Comment:
This piece is so that if a user runs `soroban contract init -f <template URL>` more than once, they won't continue to append the FE template's README to the end of their projects README. Not sure if this is an edge case that is likely to happen often, so I'm happy to rethink the inclusion of this.